src/Nethermind/Nethermind.DataMarketplace.Core/Services/GasPriceService.cs

```diff
@@ -158,6 +157,12 @@ public async Task SetPaymentClaimGasPriceAsync(UInt256 gasPrice)
 
         public async Task UpdateGasPriceAsync()
         {
+            var currentTime = _timestamper.UnixTime.Seconds;
+            if (_updatedAt + _updateInterval > currentTime || Interlocked.Exchange(ref _updatedAt, currentTime) == currentTime)
+            {
+                return;
+            }
+            
```

Comment:
This looks very odd to me. I am not sure it works as intended.

Reply:
!= current time?

Reply:
I inverted IF and I changed this condition